File: r/experiments/03_upload_and_train/run_script.R
library(azuremlsdk)

ws <- load_workspace_from_config(path = "../../../")

datastore <- get_datastore(workspace = ws, datastore_name = "blob_storage")
upload_files_to_datastore(
  datastore = datastore,
  files = list("./data/iris.csv"),
  target_path = "irisdata"
)

exp <- experiment(
  workspace = ws,
  name = "iris-from-datastore"
)
env <- r_environment(
  name = "r-environment",
  custom_docker_image = "<acr_repository_name>.azurecr.io/r-azureml:latest" # nolint
)

config <- estimator(
  source_directory = "./src",
  entry_script = "train.R",
  compute_target = "example-cluster",
  environment = env,
  script_params = list("--data_folder" = datastore$path("irisdata"))
)

run <- submit_experiment(
  experiment = exp,
  config = config
)

File: python/experiments/03_upload_and_train/run_script.py
from azureml.core import Dataset, Workspace, Experiment, ScriptRunConfig, Environment

ws = Workspace.from_config(path="../../../")

exp = Experiment(
  workspace=ws,
  name="cifar10-experiment",
)

env = ws.environments["AzureML-PyTorch-1.6-CPU"]

dataset_blob = Dataset.get_by_name(
  workspace=ws,
  name="cifar10",
)

exp_config = ScriptRunConfig(
  source_directory="./src",
  script="train.py",
  compute_target="example-cluster",
  environment=env,
  arguments=[
    "--data-path", dataset_blob.as_named_input("input").as_mount(),
    "--learning-rate", 0.005,
    "--momentum", 0.95,
  ]
)

run = exp.submit(config=exp_config)
